pkg/image/api/helper.go

```diff
@@ -290,6 +374,30 @@ func DockerImageReferenceForStream(stream *ImageStream) (DockerImageReference, e
 	return ParseDockerImageReference(spec)
 }
 
+// FollowTagReference walks through the defined tags on a stream, following any referential tags in the stream.
+// Will return ok if the tag is valid, multiple if the tag had at least reference, and ref and finalTag will be the last
+// tag seen. If a circular loop is found ok will be false.
+func FollowTagReference(stream *ImageStream, tag string) (finalTag string, ref *TagReference, ok bool, multiple bool) {
+	seen := map[string]struct{}{}
+	for {
+		tagRef, ok := stream.Spec.Tags[tag]
+		if !ok {
+			return tag, nil, false, multiple
+		}
+		if tagRef.From == nil || tagRef.From.Kind != "ImageStreamTag" || strings.Contains(tagRef.From.Name, ":") {
+			return tag, &tagRef, true, multiple
+		}
+
+		seen[tag] = struct{}{}
```

Comment:
> Is this idiomatic? Why not just use map[string]bool? I won't take memory use as an answer.

Down boy. :)

Big picture.  We all recognize this is just a `sets.String`.  My question is more focused on, "is this loop and 'exceptional' returns structure one you find easy to read".  If so, then maybe we can quibble.  If not, suggest something you think is easier.